File: explorer-frontend/src/models/Data/Block.ts
export enum BlockType {
    UNKNOWN = 0,
    POW_X16RT = 1,
    POW_ProgPow = 2,
    POW_RandomX = 3,
    POW_Sha256D = 4,
    POS = 5
}

export interface Block {
    height: number;
    hash_hex: string | null;
    strippedsize: number;
    size: number;
    weight: number;
    proof_type: BlockType;
    proofofstakehash_hex: string | null;
    progproofofworkhash_hex: string | null;
    progpowmixhash_hex: string | null;
    randomxproofofworkhash_hex: string | null;
    sha256dproofofworkhash_hex: string | null;
    proofofworkhash_hex: string | null;
    version: number;
    merkleroot_hex: string | null;
    time: number;
    mediantime: number;
    nonce: number;
    nonce64: number;
    mixhash_hex: string | null;
    bits_hex: string | null;
    difficulty: number;
    chainwork_hex: string | null;
    anon_index: number;
    veil_data_hash_hex: string | null;
    prog_header_hash_hex: string | null;
    prog_header_hex: string | null;
    epoch_number: number;
    synced: boolean;
}

File: explorer-backend/src/Models/Data/Block.cs
using ExplorerBackend.Models.Node.Response;

namespace ExplorerBackend.Models.Data;

public class Block
{
    public int height { get; set; }
    public string? hash_hex { get; set; }
    public int strippedsize { get; set; }
    public int size { get; set; }
    public int weight { get; set; }
    public BlockType proof_type { get; set; }
    public string? proofofstakehash_hex { get; set; }
    public string? progproofofworkhash_hex { get; set; }
    public string? progpowmixhash_hex { get; set; }
    public string? randomxproofofworkhash_hex { get; set; }
    public string? sha256dproofofworkhash_hex { get; set; }
    public string? proofofworkhash_hex { get; set; }
    public int version { get; set; }
    public string? merkleroot_hex { get; set; }
    public List<GetRawTransactionResult>? tx { get; set; }
    public long time { get; set; }
    public long mediantime { get; set; }
    public ulong nonce { get; set; }
    public ulong nonce64 { get; set; }
    public string? mixhash_hex { get; set; }
    public string? bits_hex { get; set; }
    public double difficulty { get; set; }
    public string? chainwork_hex { get; set; }
    public long anon_index { get; set; }
    public string? veil_data_hash_hex { get; set; }
    public string? prog_header_hash_hex { get; set; }
    public string? prog_header_hex { get; set; }
    public int epoch_number { get; set; }
    public bool synced { get; set; }
    public int txnCount { get; set; }
}

public enum BlockType : byte
{
    UNKNOWN = 0,
    POW_X16RT = 1,
    POW_ProgPow = 2,
    POW_RandomX = 3,
    POW_Sha256D = 4,
    POS = 5
}